sw/device/tests/chip_power_idle_load.c

```diff
@@ -0,0 +1,406 @@
+  };
+
+  CHECK_DIF_OK(dif_otp_ctrl_configure(&otp_ctrl
+                                     ,otp_ctrl_config
+                                     ));
+
+
+  // ------------------------------------
+  // AON Timer - activate in Watchdog mode (not wakeup mode) & IRQ
+  static const uint64_t kTimeTillBark = 1000; // time in [us] - this is to reduce VCS run time of start-to-bark to ~8 minutes
+  
+  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon_timer));      // AON timer, stop
+  CHECK_DIF_OK(dif_aon_timer_clear_wakeup_cause(&aon_timer)); // AON timer, clear
+  expected_isr_handler = wdog_irq_handler;
+  nmi_fired = false;
+  nmi_state = (dif_rv_core_ibex_nmi_state_t){0};
+  CHECK_DIF_OK(dif_rv_core_ibex_enable_nmi(&rv_core_ibex, kDifRvCoreIbexNmiSourceWdog)); // Enable WDOG NMI
+  uint32_t count_cycles = aon_timer_testutils_get_aon_cycles_from_us(kTimeTillBark); // translate to cycles
+  aon_timer_testutils_watchdog_config(&aon_timer   // AON timer, activate
+                                     ,count_cycles // the number of AON clock cycles till barking (IRQ)
+                                     ,UINT32_MAX   // the number of AON clock cycles till biting (Reset)
+                                     ,false        // false = increment even while sleeping
+                                     );
+
+  LOG_INFO("SW_INFO: AON Timer active");
+
+
+  // ------------------------------------
+  LOG_INFO("SW_INFO: all H/W is active");
+  test_status_set(0xff20); // indicate S/W is idle
+  wait_for_interrupt(); // execute "wfi" instruction
+
+  // Note, if running on DV environment:
+  // We wait here for the AON Timer to issue an NMI.
+  // This will make the S/W exit with "true" and the Verilog test to finish with "pass".
+  // Without this, the CPU will remain idle and the simulator will end the test (due to simulation timeout) with "fail".
+
+
+  // ------------------------------------
+  // Prepare to exit
+
+  // We expect the watchdog bark interrupt to be pending on the peripheral side.
+  CHECK(irq_is_pending, "Expected watchdog bark interrupt to be pending");
+  
+  // Check NMI previous state (before clearing):
+  CHECK(nmi_state.wdog_enabled && nmi_state.wdog_barked, // enabled & barked
+        "WDOG NMI state check1 not expected! wdog_enable:%x, wdog_raised:%x",
+        nmi_state.wdog_enabled, nmi_state.wdog_barked
+        );
+
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(&rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state)); // get the current state
+  // Check NMI current state (after clearing):
+  CHECK(nmi_state.wdog_enabled && !nmi_state.wdog_barked, // enabled & cleared
+        "WDOG NMI state check2 not expected! wdog_enable:%x wdog_raised:%x",
+        nmi_state.wdog_enabled, nmi_state.wdog_barked
+        );
+
+
+  // Check that no external interrupts have occurred.
+  CHECK(ext_irq_fired == false, "Unexpected external interrupt triggered.");
```

Comment:
did we actually setup the irq's to be triggered in this? I didn't immediately see the irq enables.
Regarding the NMI, is the intent to have it fire continuously? Otherwise you'll need to ack at the alert handler level, otherwise it'll keep triggering since it's considered an escalation event.

Reply:
We did not set up any external IRQs and we do not expect to get any.
Line 390 is a sanity check to make sure the CPU did not finish its WFI due to any external IRQ.
We do expect the AON Timer to issue an NMI, though.
This is a must in DV as I explained in comments inside the test.
Upon receiving the NMI, the CPU will prepare to finish the test gracefully (in terms of the Verilog simulator).
One of the things the CPU does is to shut down the AON timer.

Reply:
FYI here is a test that enables IRQs just to check at the end that they have not been triggered:
https://github.com/lowRISC/opentitan/blob/b0cdccec0f46da022ecc45cc2576cc9b9d785726/sw/device/tests/rv_core_ibex_nmi_irq_test.c#L291-L292